### PR TITLE
[10.0][ADD] product_assortment

### PR DESCRIPTION
**Pull request by rousseldenis**
Adds a method to filter to enhance it.

### CODE REVIEW

**Comment by rousseldenis** [comment]
@OCA/product-maintainers Needs review

File: product_assortment/models/ir_filters.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    record_count = fields.Integer(compute='_compute_record_count')
+
+    is_assortment = fields.Boolean(
+        default=lambda x: x._get_default_is_assortment())
+
+    @api.multi
+    def _get_eval_domain(self):
+        res = super(ProductAssortment, self)._get_eval_domain()
+
+        if self.whitelist_product_ids and res:
```
Comment (lmignon) [inline product_assortment/models/ir_filters.py:44]: @rousseldenis @Cedric-Pigeon In this code whitelist and blacklist seems to be mutually exclusive... but nothing prevents the  user to fill values for both... (A constrains ? or ...)

**Review by lmignon — CHANGES_REQUESTED** [review]
A little comment.

File: product_assortment/models/ir_filters.py (hunk 1 above)
Comment (Cedric-Pigeon) [reply product_assortment/models/ir_filters.py:44]: @lmignon why do you think they are mutually exclusive. This is not the case. The goal is to include manually some products which are not selected by the filter and/or exclude product from the filter

Comment (lmignon) [reply product_assortment/models/ir_filters.py:44]: oups sorry, I misread the code.

**Review by lmignon — APPROVED** [review]

**Comment by lmignon** [comment]
@OCA/product-maintainers  This one is ready to merge and works in production for months.

**Comment by lmignon** [comment]
No one to merge this one?

**Comment by lmignon** [comment]
Thank you @pedrobaeza !

**Comment by pedrobaeza** [comment]
You're welcome, man!

**Pull request merged**